### ISSUE TITLE
Does not work when using Reverse Proxy

### ISSUE THREAD

**Issue by leonetosoft** [RuntimeTools/appmetrics-dash#148]
Hello, I must first thank the developers for the spectacular monitoring tool.

Now I need to tell you a problem I'm having to access my metrics from an api that uses proxy.

It works as follows, my api is connected on port 5454 so I put the monitor defining the url:

 dash.attach ({
            url: '/ api / monitor'
        });
Locally worked, however external it does not bring me any monitoring data, it is empty.

My proxy works when the user types my domain name + api.

I use nginx proxy:

location / api {
                proxy_set_header X-Real-IP $ remote_addr;
                proxy_set_header X-Forwarded-For $ proxy_add_x_forwarded_for;
                proxy_set_header Host $ http_host;
                proxy_set_header X-NginX-Proxy true;
                proxy_pass http: // localhost: 5450;
                proxy_redirect off;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $ http_upgrade;
                proxy_set_header Connection "upgrade";
        }

**Comment by sjanuary** [comment]
I'm not that familiar with nginx but we do use web sockets to send data to the dash and the fact that you get the page loading but have empty graphs suggests that the web socket connection has failed.  [This page](https://www.nginx.com/blog/websocket-nginx/) suggests that you need a capital U in upgrade for websockets:
```
proxy_set_header Connection "Upgrade";
```

**Comment by leonetosoft** [comment]
That's probably it,

The dashboard is connecting to my application socket.io.
And not in the correct, I already have a socket.io running on the machine, it uses proxy:

location ~ * \ .io {
                 proxy_set_header X-Real-IP $ remote_addr;
                 proxy_set_header X-Forwarded-For $ proxy_add_x_forwarded_for;
                 proxy_set_header Host $ http_host;
                 proxy_set_header X-NginX-Proxy true;
                 proxy_pass http: // localhost: 5451;
                 proxy_redirect off;
                 proxy_http_version 1.1;
                 proxy_set_header Upgrade $ http_upgrade;
                 proxy_set_header Connection "upgrade";
          }

it is connected to port 5451.
It does not work in the same app I'm using app-metrics.

How can I make both work together?

**Comment by sjanuary** [comment]
@leonetosoft first, did you try changing `proxy_set_header Connection "upgrade";` to `proxy_set_header Connection "Upgrade";`?

**Comment by leonetosoft** [comment]
As I said,

I already have this configuration, the dashboard is connecting to my other socket server, not what it should be.

Was there something to do?

**Comment by sjanuary** [comment]
Ok, your last post was still using a lower case 'u' in Upgrade so I assumed you hadn't changed it.
The socket shouldn't connect to a different server, it uses a specific path for appmetrics-dash - see https://github.com/RuntimeTools/appmetrics-dash/blob/0518e71071378f5fff78c16c79c05cb1847b6456/public/index.html#L173 and https://github.com/RuntimeTools/appmetrics-dash/blob/45677a2cdffe017bef34597f2b02a4f532cc1320/lib/appmetrics-dash.js#L148

**Comment by leonetosoft** [comment]
Probably it should be connecting to my socket server because of this rule:

location ~* \.io {
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header Host $http_host;
                proxy_set_header X-NginX-Proxy true;
                proxy_pass http://localhost:5451;
                proxy_redirect off;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";
         }

She assumes ~ \ .io
That's why I watch him connect.

Would it be possible to change the socket port?

In case my uses 5451, I use proxy for any connection, from motorcycle I only keep the incoming port 80 open.

**Comment by leonetosoft** [comment]
'

**Comment by sjanuary** [comment]
If you use `dash.monitor` rather than `dash.attach` then it will start appmetrics-dash in a separate server on its own port.  I imagine you could then define two rules for your proxy, one for your application and one for appmetrics and different ports for each?

**Comment by leonetosoft** [comment]
I understand, using the monitor (options) is it possible to change the port that Socket.IO works?

Am I right ?

options.server = my express server, http
options.port = SocketIO Port?

Thanks for listening.

**Comment by sjanuary** [comment]
Yes that's right, you can configure the port.  It will use that port for both the dashboard web page and the socket connection.
